test/conftest.py

```diff
@@ -29,18 +29,24 @@ def file_camera_file_fixture():
     return _testdata_dir() / "FileCameraZivid2M70.zfc"
 
 
-@pytest.fixture(name="physical_camera", scope="module")
+@pytest.fixture(name="physical_camera", scope="function")
 def physical_camera_fixture(application):
     with application.connect_camera() as cam:
         yield cam
 
 
-@pytest.fixture(name="file_camera", scope="module")
+@pytest.fixture(name="file_camera", scope="function")
 def file_camera_fixture(application, file_camera_file):
     with application.create_file_camera(file_camera_file) as file_cam:
         yield file_cam
 
 
+@pytest.fixture(name="shared_file_camera", scope="module")
+def shared_file_camera_fixture(application, file_camera_file):
+    with application.create_file_camera(file_camera_file) as file_cam:
+        yield file_cam
```

Comment:
Was this done to speed things up?

Reply:
Yes. My idea is that the `shared_file_camera` can be used for any test that doesn't need to mutate the camera (for example, by disconnecting it). But if a test needs to disconnect the camera, then it cannot used the shared fixture because it may break other tests that use the same fixture. The solution for that is to use a function scoped fixture, but that makes the tests run much slower because the function scoped file camera must be re-created for each test.

I think it increases the test duration by around 2-3x.